app/frontend/components/pages/accounts/accounts.tsx

```diff
@@ -0,0 +1,286 @@
+        <a {...tooltip(`Delegate Account #${i} Stake`, true)}>
+          <span className="show-info">{''}</span>
+        </a>
+      )}
+      Delegate
+    </button>
+  )
+
+  const buttonLabel = () => {
+    if (isSelected) return 'Active'
+    if (!account) return 'Explore'
+    return 'Activate'
+  }
+
+  return (
+    <div key={i} className={`card account ${isSelected ? 'selected' : ''}`}>
+      <div className="header-wrapper mobile">
+        <h2 className="card-title small-margin">Account #{i}</h2>
+      </div>
+      <div className="card-column account-button-wrapper">
+        <h2 className="card-title small-margin account-header desktop">Account #{i}</h2>
+        <button
+          className="button primary nowrap"
+          disabled={isSelected}
+          onClick={() => {
+            setAccount(i, true)
+          }}
+        >
+          {buttonLabel()}
+        </button>
+      </div>
+      <div className="card-column account-item-info-wrapper">
+        <h2 className="card-title small-margin">Available balance</h2>
+        <div className="balance-amount small item">
+          {account ? (
+            <Balance
+              value={
+                account.shelleyBalances.stakingBalance + account.shelleyBalances.nonStakingBalance
+              }
+            />
+          ) : (
+            '-'
+          )}
+        </div>
+        <div className="mobile">
+          {account && (
+            <div className="account-action-buttons">
+              <SendFromButton />
+              <SendToButton />
+            </div>
+          )}
+        </div>
+      </div>
+      <div className="card-column account-item-info-wrapper tablet-offset">
+        <h2 className="card-title small-margin">Rewards balance</h2>
+        <div className="balance-amount small item">
+          {account ? <Balance value={account.shelleyBalances.rewardsAccountBalance} /> : '-'}
+        </div>
+      </div>
+      <div className="card-column account-item-info-wrapper">
+        <h2 className="card-title small-margin">Delegation</h2>
+        <div className="delegation-account item">
+          <PoolTicker />
+        </div>
+        <div className="mobile">
+          {account && (
+            <div className="account-action-buttons">
+              <DelegateButton />
+            </div>
+          )}
+        </div>
+      </div>
+      {account ? (
+        <div className="account-action-buttons desktop">
+          <SendFromButton />
+          <SendToButton />
+          <DelegateButton />
+        </div>
+      ) : (
+        <div className="account-action-buttons desktop" style="width: 94px;" />
+      )}
+    </div>
+  )
+}
+
+const Accounts = ({
+  accounts,
+  setAccount,
+  reloadWalletInfo,
+  showSendTransactionModal,
+  showDelegationModal,
+  shouldShowSendTransactionModal,
+  shouldShowDelegationModal,
+  selectedAccount,
+  totalWalletBalance,
+  totalRewardsBalance,
+}) => {
+  const accountInfos = Object.values(accounts)
```

Comment:
this seems fragile to me - I wouldn't take for granted that the values are always returned in ascending order, e.g. in this SO thread are contradicting answers on this: https://stackoverflow.com/questions/5525795/does-javascript-guarantee-object-property-order

Reply:
Removed.